week-2-wide-perspective/w02e02-data-manipulation/src/orders/orders.service.js

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuid } from "uuid";
+import { getRandomInt } from "../utils/index.js";
+
+export default {
+  orders: [],
+  getOrders() {
+    return fetch("http://localhost:3001/orders")
+      .then((response) => response.json())
+      .then((orders) => {
+        console.count("Counting fetch requests");
+        this.orders = orders;
+        return orders;
+      });
+  },
+  getTotalOrders(orders) {
```

Comment:
Jedyne co bym zmienił to korzystał z `this.orders` zamiast przekazywać je z zewnątrz za każdym razem.

Wiem że zrobiłeś to teraz poprawnie, w sensie - działa dużo lepiej wydajnościowo,
ale jest aż "nazbyt" elastyczne.

Super by było żeby trzymać jednak `orders` za "fasadą" i korzystać tutaj z `this.orders` w tych metodach.
Zawsze możesz, zrobić z tego klasę, w `constructor` podać wtedy bazowe `orders` i na nich działać potem w metodach.

Dlatego zamiast odbierać za każdym razem jako parametr `orders` w każdej z metod. To robiłbym to jednorazowo w jednej metodzie - `getOrders`.